scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/entity/ScheduleEntity.java

```diff
@@ -150,12 +163,19 @@ public boolean equals(Object o) {
 		if (defaultInstanceMaxCount != null ? !defaultInstanceMaxCount.equals(that.defaultInstanceMaxCount)
 				: that.defaultInstanceMaxCount != null)
 			return false;
-		if (!instanceMinCount.equals(that.instanceMinCount))
+		if (instanceMinCount != null ? !instanceMinCount.equals(that.instanceMinCount)
+				: that.instanceMinCount != null)
+			return false;
+		if (instanceMaxCount != null ? !instanceMaxCount.equals(that.instanceMaxCount)
+				: that.instanceMaxCount != null)
+			return false;

```

Comment:
instanceMinCount and instanceMaxCount are required fields so no need to do null checks, It is done for initialMinInstanceCount because initialMinInstanceCount is not required property.